addons/account/models/chart_template.py

```diff
@@ -636,6 +636,21 @@ def _post_load_data(self, template_code, company, template_data):
             company.account_purchase_tax_id = self.env['account.tax'].search([
                 *self.env['account.tax']._check_company_domain(company),
                 ('type_tax_use', 'in', ('purchase', 'all'))], limit=1).id
+        # Set default taxes on products
+        all_companies = self.env['res.company'].sudo().search([]).ids
+        # need to get all companies ids and browse the product in sudo + allowed_company_ids to check for the taxes_id and supplier_taxes_id fields 
+        # that would just look empty all the time for the current user/company
+        prod_templates = self.env['product.product'].sudo().with_context(allowed_company_ids=all_companies).search(self.env['product.product']._check_company_domain(company))
+
+        if company.account_sale_tax_id:
+            prod_templates_sale = prod_templates.filtered(
+                lambda p: p.taxes_id and not p.taxes_id.filtered_domain(p.taxes_id._check_company_domain(company)))
```

Comment:
Was explained in the commit message...

> We now assign default taxes on all existing products when creating a company, as long as it had at least already a tax set (some data products, e.g TIPS in point_of_sale, are supposed to be used without any tax)

My first try was indeed to set the default tax on all products, always, but a lot a different test cases were failing because some flows are expecting no tax at all on the product

I've make the above comment more complete